### PR TITLE
Make it possible to use `null` entities in stories

### LINKED ISSUE #8300: Erro when specifying "null" entities in stories
**Rasa version**: 2.4.2

**Rasa SDK version** 2.4.1

**Python version**: 3.7.10

**Operating system** (windows, osx, ...): Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic

**Issue**:
Rasa crashes when a story contains a "null" entity. That story would be used to handle the case where an intent is identified but an expected entity is not provided.

```
- story: test
  steps:
  - intent: test_intent
    entities:
    - test_entity: null
  - action: test_action
  ```

**Error (including full traceback)**:
```
 File "/content/rasa/bin/rasa", line 8, in <module>
    sys.exit(main())
  File "/content/rasa/lib/python3.7/site-packages/rasa/__main__.py", line 116, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/content/rasa/lib/python3.7/site-packages/rasa/cli/train.py", line 58, in <lambda>
    train_parser.set_defaults(func=lambda args: train(args, can_exit=True))
  File "/content/rasa/lib/python3.7/site-packages/rasa/cli/train.py", line 102, in train
    finetuning_epoch_fraction=args.epoch_fraction,
  File "/content/rasa/lib/python3.7/site-packages/rasa/train.py", line 109, in train
    loop,
  File "/content/rasa/lib/python3.7/site-packages/rasa/utils/common.py", line 307, in run_in_loop
    result = loop.run_until_complete(f)
  File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
  File "/content/rasa/lib/python3.7/site-packages/rasa/train.py", line 174, in train_async
    finetuning_epoch_fraction=finetuning_epoch_fraction,
  File "/content/rasa/lib/python3.7/site-packages/rasa/train.py", line 271, in _train_async_internal
    new_fingerprint = await model.model_fingerprint(file_importer)
  File "/content/rasa/lib/python3.7/site-packages/rasa/model.py", line 343, in model_fingerprint
    FINGERPRINT_STORIES_KEY: stories.fingerprint(),
  File "/content/rasa/lib/python3.7/site-packages/rasa/shared/core/training_data/structures.py", line 414, in fingerprint
    stories_as_yaml = YAMLStoryWriter().stories_to_yaml(self.story_steps)
  File "/content/rasa/lib/python3.7/site-packages/rasa/shared/core/training_data/story_writer/yaml_story_writer.py", line 119, in stories_to_yaml
    stories.append(self.process_story_step(story_step))
  File "/content/rasa/lib/python3.7/site-packages/rasa/shared/core/training_data/story_writer/yaml_story_writer.py", line 149, in process_story_step
    processed = self.process_event(event)
  File "/content/rasa/lib/python3.7/site-packages/rasa/shared/core/training_data/story_writer/yaml_story_writer.py", line 163, in process_event
    return self.process_user_utterance(event, self._is_test_story)
  File "/content/rasa/lib/python3.7/site-packages/rasa/shared/core/training_data/story_writer/yaml_story_writer.py", line 230, in process_user_utterance
    if entity["value"]:
KeyError: 'value'
```

**Command or request that led to error**:
```
rasa train
```

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- fix #8300

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/shared/core/training_data/story_writer/yaml_story_writer.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def process_user_utterance(
         if len(user_utterance.entities) and not is_test_story:
             entities = []
             for entity in user_utterance.entities:
-                if entity["value"]:
+                if "value" in entity:
```
Comment (joejuzl) [inline rasa/shared/core/training_data/story_writer/yaml_story_writer.py:230]: before this if statement was also checking if the value of `entity["value"]` was `True` right? Do we still need that behaviour?

Comment (alwx) [reply rasa/shared/core/training_data/story_writer/yaml_story_writer.py:230]: I think it was an incorrect behaviour. I've also just updated the test to check the list of entities.
What do you think?

Comment (joejuzl) [reply rasa/shared/core/training_data/story_writer/yaml_story_writer.py:230]: Yeah I think you're right - it seemed like weird behaviour.

File: tests/shared/core/training_data/story_writer/test_yaml_story_writer.py [hunk 2]
```diff
@@ -183,6 +183,33 @@ def test_yaml_writer_stories_to_yaml(domain: Domain):
     assert len(result["stories"]) == 1
 
 
+def test_yaml_writer_stories_to_yaml_with_null_entities(domain: Domain):
+    from collections import OrderedDict
```
Comment (joejuzl) [inline tests/shared/core/training_data/story_writer/test_yaml_story_writer.py:187]: This could probably be top level right?

**Review by joejuzl — APPROVED** [review]
Nice 👍

**Pull request merged**